sklearn/conftest.py

```diff
@@ -227,3 +229,76 @@ def pytest_configure(config):
         matplotlib.use("agg")
     except ImportError:
         pass
+
+
+# Definition of the random_seed fixture. See the docstring of the fixture for
+# more information.
+
+RANDOM_SEED_RANGE = list(range(100))  # All seeds in [0, 99] should be valid.
+random_seed_var = environ.get("SKLEARN_TESTS_RANDOM_SEED")
+if random_seed_var is None:
+    # If the environment variable is not defined, pick-up one seed at random in
+    # the range of admissible random seeds. Note, to make sure that all
+    # pytest-xdist workers see the same seed, we seed the meta-random number
+    # generator with a value derived from the year and the day.
+    rng = Random(int(datetime.now().strftime("%Y%j")))
+    random_seeds = [rng.choice(RANDOM_SEED_RANGE)]
+elif random_seed_var == "all":
+    random_seeds = RANDOM_SEED_RANGE
+else:
+    if "-" in random_seed_var:
+        start, stop = random_seed_var.split("-")
+        random_seeds = list(range(int(start.strip()), int(stop.strip()) + 1))
+    else:
+        random_seeds = [int(s.strip()) for s in random_seed_var.split(",")]
+
+if min(random_seeds) < 0 or max(random_seeds) > 99:
+    raise ValueError(
+        "The value(s) of the environment variable SKLEARN_TESTS_RANDOM_SEED "
+        f"must be in the range [0, 99] (or 'all'), got: {random_seed_var}"
+    )
+
+
+def pytest_report_header(config):
+    if random_seeds == RANDOM_SEED_RANGE:
+        seed_values = "all"
+    else:
+        seed_values = ",".join(str(s) for s in random_seeds)
+    return (
+        "To reproduce this test run, set the following environment variable:\n"
+        f'    SKLEARN_TESTS_RANDOM_SEED="{seed_values}"'
+    )
+
+
+@pytest.fixture(params=random_seeds)
+def random_seed(request):
```

Comment:
So as to match the convention proposed in https://github.com/scikit-learn/scikit-learn/pull/22690?
```suggestion
def global_random_seed(request):
```

Reply:
I don't think they are many existing tests where this could be a confusion so I thought the short name was better.

But I am willing to change if several reviewers prefer `global_random_seed`.

Reply:
I'm in favor of the `global` prefix. For someone new to the codebase, it makes it easier to grep for and find the definition of the fixture.

Reply:
That's a valid point.

Alternatively, I thought we could use `any_seed_in_0_99` to make it  more explicit about the content. WDYT?

Reply:
I decided to use `global_random_seed`: explicit and intuitive enough.